src/nvim/testdir/test_signs.vim

```diff
@@ -1339,3 +1710,35 @@ func Test_sign_jump_func()
   sign undefine sign1
   enew! | only!
 endfunc
+
+" Test for correct cursor position after the sign column appears or disappears.
+func Test_sign_cursor_position()
+  if !CanRunVimInTerminal()
+    throw 'Skipped: cannot make screendumps'
+  endif
+
+  let lines =<< trim END
+	call setline(1, [repeat('x', 75), 'mmmm', 'yyyy'])
+	call cursor(2,1)
+   	sign define s1 texthl=Search text==>
+	redraw
+   	sign place 10 line=2 name=s1
+  END
```

Comment:
could add this as screen test in `test/functional/ui/sign_spec.lua`

Reply:
I prefer not to port it now until the other sign patches that touch this test are merged.

I can decouple the restricted and sign patches to separate PRs if the rest are good to go.